core/rawdb/database_test.go

```diff
@@ -0,0 +1,136 @@
+package rawdb
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/libevm/ethdb"
+)
+
+func ExampleInspectDatabase() {
+	extendRight := func(base []byte, totalLength int) []byte {
+		if len(base) > totalLength {
+			return base[:totalLength]
+		}
+		extended := make([]byte, totalLength)
+		copy(extended, base)
+		return extended
+	}
+
+	db := &stubDatabase{
+		iterator: &stubIterator{
+			kvs: []keyValue{
+				// Extra metadata keys: (17 + 1) + (9 + 1) = 28 bytes
+				{key: snapshotBlockHashKey, value: []byte("a")},
+				{key: syncRootKey, value: []byte("b")},
+				// Trie segments: 77 + 1 = 78 bytes
+				{key: extendRight(syncSegmentsPrefix, syncSegmentsKeyLength), value: []byte("c")},
+				// Storage tries to fetch: 76 + 1 = 77 bytes
+				{key: extendRight(syncStorageTriesPrefix, syncStorageTriesKeyLength), value: []byte("d")},
+				// Code to fetch: 34 + 1 = 35 bytes
+				{key: extendRight(CodeToFetchPrefix, codeToFetchKeyLength), value: []byte("e")},
+				// Block numbers synced to: 22 + 1 = 23 bytes
+				{key: extendRight(syncPerformedPrefix, syncPerformedKeyLength), value: []byte("f")},
```

Comment:
hmm I wonder if we should directly use `rawdb` methods like `rawdb.AddCodeToFetch`, `rawdb.WriteSyncSegment` etc?

Reply:
Indeed, done 👍 !